Mage/src/main/java/mage/abilities/Ability.java

```diff
@@ -34,10 +34,15 @@
 public interface Ability extends Controllable, Serializable {
 
     /**
-     * Assigns a new {@link java.util.UUID}
+     * Assigns a new random {@link java.util.UUID}
```

Comment:
Also xmage uses time based effect's order like paper rules. Maybe it can be bugged with dependency effects (if some effect depend on another then it must be applied later). Example: effect that checking card type must be applied after effect that change card type. Search for `dependencyTypes` and `dependendToTypes`

Reply:
The death doesn't happen in the middle of a calculation, it happens after the continuous effects were applied in the "wrong" order (massive quotes on "wrong" here because the order shouldn't matter). The problem is that there are both "set toughness to 0" and "set toughness to 5" effects ok the same sublayer affecting the same creature, resulting in what cosmetically could be compared to a race condition.

Reply:
> Also xmage uses time based effect's order like paper rules (due effects adding timestamp). Maybe it can be bugged with dependency effects (if some effect depend on another then it must be applied later). Example: effect that checking card type must be applied after effect that change card type. Search for `dependencyTypes` and `dependendToTypes`

I'll check this out. Maybe my understanding of effects on the same sublayer is wrong

Reply:
See details in rule 613.8:
> 613.8. Within a layer or sublayer, determining which order effects are applied in is sometimes done using a dependency system. If a dependency exists, it will override the timestamp system. 

XMage support it but it require to manual setup (see `DependencyType` usage), so only few popular use cases implemented. Rare or new effects/combo can be miss.

Reply:
The "set toughness to 5" effect should systematically always apply after the "set toughness to 0" effect (by timestamp) even with both in layer 7b. If that can't be easily fixed, then there's no good reason for the "set toughness to 0" effect to be here at all in this specific card application.

(Maybe there are underlying issues with dependencies but that's not directly related.)